### PR TITLE
Acceptance: remove has_error functions from all the tests  

### PR DESCRIPTION
**Pull request by rikirenz**
## Description
This PR wants to remove the `has_error` function that we used in the acceptance test.

## Motivation and Context
The has_error function was confusing because was not used in the proper way. With this change in the `ArsenicResponse` class we are able to decide which kind of assertion we want to test plus the naming with this change is going to be more meaningful

## Checklist:
- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rikirenz** [comment]
@kaplun This meta-programming IMHO is necessary in order to avoid that the tests use directly selenium. In this way with the `ArsenicRepsonse` the test does not have to care about the selenium framework, as result we have: 
- The test is able to self document itself because is more readable
- We can reuse more code

😄

File: inspirehep/bat/arsenic.py [hunk 1]
```diff
@@ -72,7 +72,15 @@ def __getattr__(self, item):
 
 
 class ArsenicResponse(object):
-    has_error = None
+    statement = None
+    negative_statement = None
 
-    def __init__(self, has_error):
-        self.has_error = has_error
+    def __init__(self, statement, negative_statement=None):
```
Comment (david-caro) [inline inspirehep/bat/arsenic.py]: We should add here some docs on how to use this, and what the params mean.

Comment (rikirenz) [inline inspirehep/bat/arsenic.py]: Added :fist_oncoming:

**Comment by jacquerie** [comment]
This PR is trading one antipattern for two antipatterns. We didn't need `ArsenicResponse` and the `has_error` mechanism in the first place, since they wrap what would have been, if written semantically, `assert` statements.

**Comment by david-caro** [comment]
@jacquerie please, provide an alternative, I still think that this PR improves the current status of the code by moving the assert to the point where it actually does the check if the page was properly loaded, and helping then with the debugging of the issue (currently, you end up in a context where you are just checking if `assert True` or `assert False`, not really helping to debug why).

**Comment by jacquerie** [comment]
Yes, it's bad, but this PR goes deeper in the bad direction: instead of simply writing assertions like `assert 'this text' in this_field` these tests had the complicated mechanism of `has_error` and closures and what not, and this PR makes that even more complicated by adding another similar mechanism. Really, the only alternative I see here is to slowly rewrite these tests one by one.

**Comment by david-caro** [comment]
The problem there is that the assertion (or assertions) that need to be made depend on the page you are in, and that information is in the domain of the bat pages. What the `ArsenicResponse` does is to define an interface that all the pages have to implement to unify that `assert` at the test level.

I don't think that the idea of such interface is wrong, I think that it's actually good (abstracting the tests from the individual nuances of each page, you don't really know if having the string `alert_wrn_smthng` is actually an error or a success, also all the xpath and similar things get "hidden" under a function improving the readability of the whole test). Just that the current implementation is lacking expressiveness (though I think that this PR also improves that point too, instead of the ambiguous `has_error` it now has `assert_errors` and `assert_no_errors` that actually match what they do).

About the passing of closures, it can be done also with subclassing, though it will increase the boilerplate needed (having to add a new `ArsenicResponse` subclass to each page).

**Pull request closed without merging**